### PR TITLE
Getting Started with Recoil as a State Management Library for React Applications

### LINKED ISSUE #4792: Getting Started with Recoil as a State Management Library for React Applications
#### Getting Started with Recoil as a State Management Library for React Applications

Managing your data in frontend frameworks is equally important as in databases. It helps avoid making many API calls and reduces passing too many props from one component to the other. When you have state management in place data actually flows from your app to state and vice versa. You know exactly where your data is. These state management tools also give you a point-in-time snapshot of the entire data. In that way, you know exactly where your data is and that makes your development faster.

In this article, we will see how we can manage your data well with Recoil (a state management library for react) in React applications. This library is beginner-friendly as it is much easier to get started with compared to other state management libraries like Redux.

### Pre-requisites
- Basic understanding of React.
- Understanding of React functional components.
- Understanding of React hooks.

#### Key takeaways 
- Understanding the concept of state management.
- Getting started with Recoil.
- Understanding how to manage state in your React applications.  

#### Article quality
My article is unique in the following ways:
- The other articles on the internet assume that the reader has previous experience with state management tools like Redux. My article will explain everything from the basics.
- I will use an example that will help the reader get started with any kind of project.

### PR DESCRIPTION
**Pull request by annemwangi**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4792 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Review by jumamiller — COMMENTED** [review]
At glance, everything looks good. I'll review as soon as possible!

**Comment by LinusMuema** [comment]
Hi @annemwangi 

Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept. Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content. Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[Getting Started with Recoil as a State Management Library for React Applications #5133 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7608242/Getting.Started.with.Recoil.as.a.State.Management.Library.for.React.Applications.5133.Copyleaks.Report.pdf)

**Comment by annemwangi** [comment]
Hello @LinusMuema, the plagiarism is because I copied the introduction from the topic suggestion. I'll fix it ASAP and let you know...

**Comment by annemwangi** [comment]
Revisions done @LinusMuema  🚀

File: content/authors/anne-mwangi/index.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Anne Mwangi is an undergraduate student pursuing Bachelor of Science in Computer Science. She is a lover of the web and has specialized in the JavaScript eco-system. She likes solving problems and she is a critical thinker.
```
Comment (jumamiller) [inline content/authors/anne-mwangi/index.md:1]: Hi, kindly reformat the author details following the set format!

**Comment by hectorkambow** [comment]
👋 Good afternoon - and welcome to the EngEd program.
@annemwangi 
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

**Comment by hectorkambow** [comment]
[Getting Started with Recoil as a State Management Library for React Applications #5133.pdf](https://github.com/section-engineering-education/engineering-education/files/7652614/Getting.Started.with.Recoil.as.a.State.Management.Library.for.React.Applications.5133.pdf)

**Comment by annemwangi** [comment]
On it...

**Comment by annemwangi** [comment]
> 👋 Good afternoon - and welcome to the EngEd program. @annemwangi Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept. Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content. Be sure to see our resources page to see more info on plagiarism and what is considered as such.

Hello @hectorkambow, thanks for welcoming me to the platform. The plagiarism was mostly because of the code snippets. I have done some major changes and run my article through a plagiarism checker, [duplichecker](https://www.duplichecker.com/) and it shows that the article has no plagiarism now. I have no access to premium tools.

**Comment by jumamiller** [comment]
> > wave Good afternoon - and welcome to the EngEd program. @annemwangi Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept. Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content. Be sure to see our resources page to see more info on plagiarism and what is considered as such.
> 
> Hello @hectorkambow, thanks for welcoming me to the platform. The plagiarism was mostly because of the code snippets. I have done some major changes and run my article through a plagiarism checker, [duplichecker](https://www.duplichecker.com/) and it shows that the article has no plagiarism now. I have no access to premium tools.

Hi, just go through the attached document(above) and fix the areas causing plagiarism. These tools defer in accuracy!

**Comment by annemwangi** [comment]
> > > wave Good afternoon - and welcome to the EngEd program. @annemwangi Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept. Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content. Be sure to see our resources page to see more info on plagiarism and what is considered as such.
> > 
> > 
> > Hello @hectorkambow, thanks for welcoming me to the platform. The plagiarism was mostly because of the code snippets. I have done some major changes and run my article through a plagiarism checker, [duplichecker](https://www.duplichecker.com/) and it shows that the article has no plagiarism now. I have no access to premium tools.
> 
> Hi, just go through the attached document(above) and fix the areas causing plagiarism. These tools defer in accuracy!

I already did that @jumamiller, the remaining one are things that you just can't change e.g., `import { atom, selector } from 'recoil';`. You can't change them because its the predefined syntax of doing  it.

**Comment by jumamiller** [comment]
Hi @annemwangi, kindly rename your author image to "avatar..." and not hero.

**Comment by jumamiller** [comment]
> Hi @annemwangi, kindly rename your author image to "avatar..." and not hero.

and then update it on your author page!

**Comment by Kayere** [comment]
Plagiarism check done.

**Review by jumamiller — APPROVED** [review]
Approved!

**Comment by hectorkambow** [comment]
Overall nice work - easy to follow along tutorial

**Pull request merged**